### PR TITLE
Show migration history

### LINKED ISSUE #19: Show migration history in UI
this brings back the init message, see

```tsx
<pre className="mb-0">{JSON.stringify({} /* TODO re-implement */, null, "  ")}</pre>
```

The information is available in `CosmWasmClient.getContractCodeHistory`

### PR DESCRIPTION
**Pull request by abefernan**
Closes #19

### CODE REVIEW

**Review by webmaster128 — COMMENTED** [review]
Good start. Let's create a dedicate UI component for each

```
// An entry in the contracts code/ migration history
export interface ContractCodeHistoryEntry {
  /** The source of this history entry */
  readonly operation: "Genesis" | "Init" | "Migrate";
  readonly code_id: number;
  readonly msg: Record<string, unknown>;
}
```

and only show the `msg` as raw JSON. This allows nice think like being able to link to older code IDs

**Comment by abefernan** [comment]
> and only show the `msg` as raw JSON. This allows nice think like being able to link to older code IDs

@webmaster128 I just pushed this UI:
image

File: src/pages/contract/ContractPage.tsx [hunk 1]
```diff
@@ -139,7 +145,10 @@ export function ContractPage(): JSX.Element {
             ) : isErrorState(details) ? (
               <p>An Error occurred when loading contract</p>
             ) : (
-              <InitializationInfo contract={details} instantiationTx={instantiationTx} />
+              <>
```
Comment (webmaster128) [inline src/pages/contract/ContractPage.tsx:148]: Is this a shortcut for `<Fragment>`?

File: src/pages/contract/ContractPage.tsx [hunk 2]
```diff
@@ -88,6 +92,8 @@ export function ContractPage(): JSX.Element {
         setInstantiationTx(first);
       })
       .catch(() => setInstantiationTx(errorState));
+
+    clientContext.client.getContractCodeHistory(contractAddress).then(setContractCodeHistory);
```
Comment (webmaster128) [inline src/pages/contract/ContractPage.tsx:96]: Here we should at least catch and log errors with:

```ts
      .catch((error) => {
        console.error(error);
      });
```

No need for a UI now but then we can easily add one

File: src/pages/contract/HistoryInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export function HistoryInfo({ contractCodeHistory }: Props): JSX.Element {
+  return (
+    <div className="card mb-3">
+      <ul className="list-group list-group-flush">
+        <li className="list-group-item">
+          <span title="The contract level message history">History</span>
+        </li>
+        {contractCodeHistory.map((entry) => (
+          <li className="list-group-item">
```
Comment (webmaster128) [inline src/pages/contract/HistoryInfo.tsx:18]: Isn't React complaining about missing `key=` here?

**Review by webmaster128 — COMMENTED** [review]
Beautiful

File: src/pages/contract/ContractPage.tsx (hunk 1 above)
Comment (abefernan) [reply src/pages/contract/ContractPage.tsx:148]: Yes

File: src/pages/contract/HistoryInfo.tsx (hunk 3 above)
Comment (abefernan) [reply src/pages/contract/HistoryInfo.tsx:18]: Oops, yes it is. But I don't know what to use as a unique key from `ContractCodeHistoryEntry` properties

Comment (webmaster128) [reply src/pages/contract/HistoryInfo.tsx:18]: You can use the index via: `contractCodeHistory.map((entry, index) => (`

**Review by webmaster128 — APPROVED** [review]

**Pull request merged**